src/Logging/HTTPOutputHandler.php

```diff
@@ -145,6 +146,13 @@ protected function write(array $record)
     {
         ini_set('display_errors', 0);
 
+        if (Deprecation::isLoggingOnly() && isset($record['context']['code'])) {
+            $errorCode = $record['context']['code'];
+            if ($errorCode === E_USER_DEPRECATED || $errorCode === E_DEPRECATED) {
+                return;
+            }
+        }
```

Comment:
In this context we're going to be adding output to either an HTTP response or a CLI response. Either way, we only want to output deprecation notices if `isLoggingOnly()` is false.